fedimint-client/src/lib.rs

```diff
@@ -624,12 +638,50 @@ impl Client {
     }
 
     pub async fn get_operation(&self, operation_id: OperationId) -> Option<OperationLogEntry> {
-        self.inner
-            .db
-            .begin_transaction()
+        Client::get_operation_inner(&mut self.inner.db.begin_transaction().await, operation_id)
             .await
-            .get_value(&OperationLogKey { operation_id })
+    }
+
+    async fn get_operation_inner(
+        dbtx: &mut DatabaseTransaction<'_>,
+        operation_id: OperationId,
+    ) -> Option<OperationLogEntry> {
+        dbtx.get_value(&OperationLogKey { operation_id }).await
+    }
+
+    /// Sets the outcome of an operation
+    #[instrument(skip(db), level = "debug")]
+    pub async fn set_operation_outcome(
+        db: &Database,
+        operation_id: OperationId,
+        outcome: &(impl Serialize + Debug),
+    ) -> anyhow::Result<()> {
+        let outcome_json = serde_json::to_value(outcome).expect("Outcome is not serializable");
+
+        let mut dbtx = db.begin_transaction().await;
+        let mut operation = Client::get_operation_inner(&mut dbtx, operation_id)
             .await
+            .expect("Operation exists");
+        operation.outcome = Some(outcome_json);
+        dbtx.insert_entry(&OperationLogKey { operation_id }, &operation)
+            .await;
+        dbtx.commit_tx_result().await.expect("commit failed");
```

Comment:
Can you use `commit_tx` instead of `commit_tx_result`? Can remove the expect that way